src/main.rs

```diff
@@ -13,6 +13,10 @@ fn main() -> Result<()> {
 
     match cli::parse_arguments(&raw_args)? {
         CliCommand::Benchmark { command } => benchmark(&command),
-        CliCommand::Show => show(),
+        CliCommand::Show { row, col, metric } => match (row, col, metric) {
+            (Some(row), Some(col), Some(metric)) => show_2d_table(row, col, metric),
+            (Some(row), _, Some(metric)) => show_1d_table(row, metric),
+            _ => show(),
```

Comment:
FYI: A way to make this a bit more easy to understand, you could also split CliCommand::Show in `CliCommand::Show`, `CliComnmand::Show1dTable` and `CliCommand::Show2dTable`. Therefore only specific permutations of `row`, `col` and `metric` are allowed by construction. But it is OK as you implemented it, because the argument parsing should limit the allowed permutations anyway.

Reply:
Indeed I also considered this kind of solution but wasn’t sure if it’s better or not atm 🤔 .